20 Goals Operations  with MongoDB Integration

Qloik opened the pull request:
I added create/delete/update/get operations for goals and they can be saved into local  MongoDB database

RESTful API endpoints for goals:
  1. POST /api/goals - Create a new goal
  2. GET /api/goals/id - Retrieve a specific goal
  3. PUT /api/goals/id - Update a goal
  4. DELETE /api/goals/id - Delete a goal

### Sample Goal Schema:
```body-> raw-> json
{
    "title": "Hi, I'm apple",
    "description": "no you are not",
    "deadline": "2024-03-01T00:00:00.000Z"
}

aaron-wu1 commented on backend/src/controllers/goalController.ts:
```diff
@@ -0,0 +1,149 @@
+export const updateGoal = async (req: Request, res: Response) => {
+    try {
+        const goalId = req.params.id;
+        const updates = req.body;
```
Might be nice to take advantage of typing here for the request body to know what kinds of expected inputs are allowed or a comment would be nice

Qloik replied:
Thank you for the suggestion! I'll add an interface to define the expected update fields and their types to make the allowed inputs clearer.

aaron-wu1 commented on backend/src/types/express/index.d.ts:
```diff
@@ -2,8 +2,8 @@ declare namespace Express {
     export interface Request {
```
Looking at the folder hierarchy, what I'm assuming here is the file where the interfaces are going to be declared for the whole api. Wondering if this is extensible or not, given the future features.

Qloik replied:
Yeah I will check this files and try to maintain consistency.

aaron-wu1 reviewed (commented):
Left some clarifying comments

aaron-wu1 commented:
I realized that this is not on the schema on figma, but there's a boolean value to for completing a goal that was in the UI figma. Is it possible to add this in also?

Qloik commented:
@aaron-wu1 @ziruichen2020 I've updated the goal model to use isCompleted boolean instead of status enum, and also cleaned up the test cases accordingly. Btw .DS_Store files have been removed from tracking. Please review and merge. :)
